core/src/main/java/greencity/service/impl/UserServiceImpl.java

```diff
@@ -925,9 +925,30 @@ public UserAndAllFriendsWithOnlineStatusDto getAllFriendsWithTheOnlineStatus(Lon
     /**
      * {@inheritDoc}
      */
+    @Transactional
     @Override
     public void deactivateUser(Long id) {
         User foundUser = findById(id);
         foundUser.setUserStatus(UserStatus.DEACTIVATED);
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Transactional
+    @Override
+    public List<Long> deactivateAllUsers(List<Long> listId) {
+        listId.forEach(this::deactivateUser);
```

Comment:
There is a problem with this approach. What if in the middle of the list a user is not found? It will fail in the middle, does it mean that some users from the list will be deactivated and some not?

Reply:
Changed approach.